### PR TITLE
Hash with Indifferent access

### PR DESCRIPTION
**Pull request by T-Dnzt**
This PR add an automatic transformation of any hash going through a step to `HashWithIndifferentAccess`. Transformers parameters that are being accessed by record key are also being converted to HashWithIndifferentAccess at initialization.

These changes make it easier for anyone to work with Optimus Prime and build a pipeline since records can now be accessed with string and symbols. A bit of caution is still required for the transformers using hashes to interact with the records. 

Note that `HashWithIndifferentAccess` does not play nicely with keyword arguments. (Basically, it crashes :smiling_imp: )

### CODE REVIEW

**Comment by omarkhan** [comment]
:+1: woop woop

**Comment by simonbnrd** [comment]
:+1:

**Comment by jorin-vogel** [comment]
:+1:

File: lib/optimus_prime/transformers/change_value.rb [hunk 1]
```diff
@@ -18,7 +18,7 @@ module Transformers
     class ChangeValue < Destination
       # value_map - Hash of existing values to transform to new values
       def initialize(value_map: {})
-        @value_map = value_map
+        @value_map = value_map.with_indifferent_access
```
Comment (TamerShlash) [inline lib/optimus_prime/transformers/change_value.rb]: Why are we making config `with_different_access` if we are already doing the data with indifferent access. If so, we would just do `indefferent_access` for the config and leave that data alone. What do you think @T-Dnzt @omarkhan?

**Comment by iammchat** [comment]
:+1:

File: lib/optimus_prime/transformers/change_value.rb (hunk 1 above)
Comment (omarkhan) [inline lib/optimus_prime/transformers/change_value.rb]: I think it makes sense to use indifferent access for the data. I don't really mind what we do for the config

Comment (T-Dnzt) [inline lib/optimus_prime/transformers/change_value.rb]: As discussed, the keyword attributes we're using in most method definitions is incompatible with `HashWithIndifferentAccess`. I probably put `with_indifferent_access` in some places where we don't actually need it but it's mostly to keep things uniform and prevent problems in the future if someone tries to use symbols or strings to access some config hash.

**Comment by TamerShlash** [comment]
:+1:

**Comment by Anumodana** [comment]
:smiley: :+1:

**Pull request merged**